streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -150,10 +150,10 @@ public void handleAssignment(final Map<TaskId, Set<TopicPartition>> activeTasks,
             } else /* we previously owned this task, and we don't have it anymore, or it has changed active/standby state */ {
                 final Set<TopicPartition> inputPartitions = task.inputPartitions();
                 try {
-                    task.closeClean();
                     changelogReader.remove(task.changelogPartitions());
+                    task.closeClean();
                 } catch (final RuntimeException e) {
-                    log.error("Failed to close task {} cleanly. Attempting to close remaining tasks before re-throwing.", task.id());
+                    log.error("Failed to close task {} cleanly. Attempting to close remaining tasks before re-throwing:", task.id(), e);
```

Comment:
This is one of Java's best-laid traps. You can't mix "format args" mode and "record exception" mode. The line as written will actually just not print the exception at all.
```suggestion
                    log.error(String.format("Failed to close task %s cleanly. Attempting to close remaining tasks before re-throwing:", task.id()), e);
```

Reply:
Fooled me I guess